src/stdlib_experimental_io.f90

```diff
@@ -268,4 +275,108 @@ logical function whitechar(char) ! white character
 end if
 end function
 
+integer function open(filename, mode) result(u)
+! Open a file
+!
+! To open a file to read:
+!
+! u = open("somefile.txt")        # The default `mode` is "rt"
+! u = open("somefile.txt", "r")
+!
+! To open a file to write:
+!
+! u = open("somefile.txt", "w")
+
+! To append to the end of the file if it exists:
+!
+! u = open("somefile.txt", "a")
+
+character(*), intent(in) :: filename
+character(*), intent(in), optional :: mode
+integer :: io
+character(3):: mode_
+character(:),allocatable :: action_, position_, status_, access_, form_
+
+
+mode_ = parse_mode(optval(mode, ""))
+
+if (mode_(1:2) == 'r ') then
+    action_='read'
+    position_='asis'
+    status_='old'
+else if (mode_(1:2) == 'w ') then
+    action_='write'
+    position_='asis'
+    status_='replace'
+else if (mode_(1:2) == 'a ') then
+    action_='write'
+    position_='append'
+    status_='old'
+else if (mode_(1:2) == 'x ') then
+    action_='write'
+    position_='asis'
+    status_='new'
+else if (mode_(1:2) == 'r+') then
+    action_='readwrite'
+    position_='asis'
+    status_='old'
+else if (mode_(1:2) == 'w+') then
+    action_='readwrite'
+    position_='asis'
+    status_='replace'
+else if (mode_(1:2) == 'a+') then
+    action_='readwrite'
+    position_='append'
+    status_='old'
+else if (mode_(1:2) == 'x+') then
+    action_='readwrite'
+    position_='asis'
+    status_='new'
+else
+    call error_stop("Unsupported mode: "//mode_(1:2))
+end if
+
+if (mode_(3:3) == 't') then
+    access_='sequential'
+    form_='formatted'
+else if (mode_(3:3) == 'b' .or. mode_(3:3) == 's') then
+    access_='stream'
+    form_='unformatted'
+else
+    call error_stop("Unsupported mode: "//mode_(3:3))
+endif
+
+open(newunit=u, file=filename, &
+     action = action_, position = position_, status = status_, &
+     access = access_, form = form_, &
+     iostat = io)
+
+end function
+
+character(3) function parse_mode(mode) result(mode_)
+character(*), intent(in) :: mode
+
+mode_ = 'r t'
+if (len_trim(mode) == 0) return
+mode_(1:1) = mode(1:1)
+
+if (len_trim(adjustl(mode)) > 1) then
+    if (mode(2:2) == '+' )then
+        mode_(2:2) = '+'
+    else
+        mode_(3:3) = mode(2:2)
+    endif
+end if
+
+if (len_trim(adjustl(mode)) > 2) then
+        mode_(3:3) = mode(3:3)
+end if
+
+if (mode_(1:1) == 'b') then
+    mode_(1:1) = mode_(3:3)
+    mode_(3:3) = 'b'
+end if
+
```

Comment:
Something like these lines should also appear for 't'. I modified this function to allow all cases. 
Anyway, for now, I propose to accept the PR as is, and I will then submit a new PR to modify this function.

Reply:
==>Issue solved if PR #77 is accepted

Reply:
Merged.